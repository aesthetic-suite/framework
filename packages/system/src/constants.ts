import { ColorShade, ElevationType, PaletteType } from './types';

export const DEPTHS: Record<ElevationType, number> = {
  content: 100, // xs
  navigation: 1000, // sm
  sheet: 1100, // lg
  overlay: 1200, // lg
  modal: 1300, // xl
  toast: 1400, // md
  dialog: 1500, // md
  menu: 1600, // md
  tooltip: 1700, // sm
};

export const SHADE_RANGES: ColorShade[] = [
  '00',
  '10',
  '20',
  '30',
  '40',
  '50',
  '60',
  '70',
  '80',
  '90',
];

export const PALETTE_TYPES: PaletteType[] = [
  'brand',
  'primary',
  'secondary',
  'tertiary',
  'neutral',
  'muted',
  'danger',
  'warning',
  'negative',
  'positive',
];
