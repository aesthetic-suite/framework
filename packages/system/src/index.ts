/**
 * @copyright   2017-2019, Miles Johnson
 * @license     https://opensource.org/licenses/MIT
 */

import Design from './Design';
import Theme from './Theme';

export * from './types';

export { Design, Theme };
