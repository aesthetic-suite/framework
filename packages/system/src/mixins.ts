import { objectLoop } from '@aesthetic/utils';
import { MixinTemplate, MixinType, Utilities } from './types';

export function hideOffscreen() {
	return {
		clipPath: 'rect(1px, 1px, 1px, 1px)',
		height: 1,
		left: '-5vw',
		overflow: 'hidden',
		position: 'fixed',
		width: 1,
	};
}

export function hideVisually() {
	return {
		'@selectors': {
			':not(:focus):not(:active)': {
				border: 0,
				clip: 'rect(0, 0, 0, 0)',
				clipPath: 'inset(50%)',
				height: 1,
				margin: 0,
				overflow: 'hidden',
				padding: 0,
				position: 'absolute',
				whiteSpace: 'nowrap',
				width: 1,
			},
		},
	};
}

export function resetButton() {
	return {
		appearance: 'none',
		backgroundColor: 'transparent',
		border: 0,
		cursor: 'pointer',
		display: 'inline-flex',
		fontSize: 'inherit',
		margin: 0,
		padding: 0,
		textDecoration: 'none',
		userSelect: 'auto',
		verticalAlign: 'middle',
	};
}

export function resetInput() {
	return {
		appearance: 'none',
		backgroundColor: 'transparent',
		margin: 0,
		padding: 0,
		width: '100%',
		'@selectors': {
			'::-moz-focus-outer': {
				border: 0,
			},
		},
	};
}

export function resetList() {
	return {
		listStyle: 'none',
		margin: 0,
		padding: 0,
	};
}

export function resetMedia() {
	return {
		display: 'block',
		verticalAlign: 'middle',
	};
}

export function resetTypography() {
	return {
		fontFamily: 'inherit',
		fontSize: 'inherit',
		fontWeight: 'normal',
		wordWrap: 'break-word',
	};
}

export function root(css: Utilities<object>) {
	const declaration = {
		backgroundColor: css.var('palette-neutral-color-00'),
		color: css.var('palette-neutral-text'),
		fontFamily: css.var('typography-font-text'),
		fontSize: css.var('typography-root-text-size'),
		lineHeight: css.var('typography-root-line-height'),
		textRendering: 'optimizeLegibility',
		textSizeAdjust: '100%',
		margin: 0,
		padding: 0,
		'-webkit-font-smoothing': 'antialiased',
		'-moz-osx-font-smoothing': 'grayscale',
		'@media': {},
	};

	// Fluid typography!
	objectLoop(css.tokens.breakpoint, (bp, size) => {
		(declaration['@media'] as Record<string, object>)[bp.query] = {
			fontSize: css.var(`breakpoint-${size}-root-text-size`),
			lineHeight: css.var(`breakpoint-${size}-root-line-height`),
		};
	});

	return declaration;
}

export function textBreak() {
	return {
		overflowWrap: 'break-word',
		wordWrap: 'break-word',
		wordBreak: 'break-word',
	};
}

export function textTruncate() {
	return {
		overflow: 'hidden',
		textOverflow: 'ellipsis',
		whiteSpace: 'nowrap',
	};
}

export function textWrap() {
	return {
		overflowWrap: 'normal',
		wordWrap: 'normal',
		wordBreak: 'normal',
	};
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const MIXIN_MAP: Record<MixinType, MixinTemplate<any>> = {
	'hide-offscreen': hideOffscreen,
	'hide-visually': hideVisually,
	'reset-button': resetButton,
	'reset-input': resetInput,
	'reset-list': resetList,
	'reset-media': resetMedia,
	'reset-typography': resetTypography,
	root,
	'text-break': textBreak,
	'text-truncate': textTruncate,
	'text-wrap': textWrap,
};
