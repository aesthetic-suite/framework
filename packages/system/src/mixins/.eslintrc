{
  "rules": {
    "babel/no-invalid-this": "off"
  }
}
