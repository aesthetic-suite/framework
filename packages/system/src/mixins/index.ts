export * from './background';
export * from './border';
export * from './display';
export * from './foreground';
export * from './heading';
export * from './reset';
export * from './root';
export * from './shadow';
export * from './text';
export * from './ui';
