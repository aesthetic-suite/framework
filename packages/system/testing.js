module.exports = require('./lib/testing.js');
