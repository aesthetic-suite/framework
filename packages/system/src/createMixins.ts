import { objectCreate } from '@aesthetic/utils';
import mixins from './mixins';
import { Mixins, VarUtil, Tokens } from './types';
import { BORDER_SIZES, HEADING_SIZES, SHADOW_SIZES, TEXT_SIZES, PALETTE_TYPES } from './constants';

export default function createMixins(
  vars: VarUtil,
  // Only ask for what we need
  tokens: {
    breakpoint: Tokens['breakpoint'];
  },
): Mixins {
  return {
    background: objectCreate(PALETTE_TYPES, (palette) => mixins.background(vars, palette)),
    border: objectCreate(BORDER_SIZES, (size) => mixins.border(vars, size)),
    box: objectCreate(TEXT_SIZES, (size) => mixins.box(vars, size)),
    foreground: objectCreate(PALETTE_TYPES, (palette) => mixins.foreground(vars, palette)),
    heading: objectCreate(HEADING_SIZES, (size) => mixins.heading(vars, size)),
    pattern: {
      hide: {
        completely: mixins.hideCompletely(),
        offscreen: mixins.hideOffscreen(),
        visually: mixins.hideVisually(),
      },
      reset: {
        button: mixins.resetButton(),
        input: mixins.resetInput(),
        list: mixins.resetList(),
        media: mixins.resetMedia(),
        typography: mixins.resetTypography(),
      },
      root: mixins.root(vars, tokens.breakpoint),
      text: {
        break: mixins.textBreak(),
        truncate: mixins.textTruncate(),
        wrap: mixins.textWrap(),
      },
    },
    shadow: objectCreate(SHADOW_SIZES, (size) => mixins.shadow(vars, size)),
    text: objectCreate(TEXT_SIZES, (size) => mixins.text(vars, size)),
    ui: {
      button: objectCreate(PALETTE_TYPES, (palette) => mixins.button(vars, palette)),
    },
  };
}
