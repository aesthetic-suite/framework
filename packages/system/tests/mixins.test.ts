import mixins from '../src/mixins';
import { darkTheme } from '../src/testing';

describe('Mixins', () => {
  it('border', () => {
    expect(mixins.border(darkTheme.var, 'sm')).toMatchSnapshot();
    expect(mixins.border(darkTheme.var, 'df')).toMatchSnapshot();
    expect(mixins.border(darkTheme.var, 'lg')).toMatchSnapshot();
  });

  it('box', () => {
    expect(mixins.box(darkTheme.var, 'sm')).toMatchSnapshot();
    expect(mixins.box(darkTheme.var, 'df')).toMatchSnapshot();
    expect(mixins.box(darkTheme.var, 'lg')).toMatchSnapshot();
  });

  it('heading', () => {
    expect(mixins.heading(darkTheme.var, 'l1')).toMatchSnapshot();
    expect(mixins.heading(darkTheme.var, 'l2')).toMatchSnapshot();
    expect(mixins.heading(darkTheme.var, 'l3')).toMatchSnapshot();
    expect(mixins.heading(darkTheme.var, 'l4')).toMatchSnapshot();
    expect(mixins.heading(darkTheme.var, 'l5')).toMatchSnapshot();
    expect(mixins.heading(darkTheme.var, 'l6')).toMatchSnapshot();
  });

  it('pattern', () => {
    expect(mixins.hideCompletely()).toMatchSnapshot();
    expect(mixins.hideOffscreen()).toMatchSnapshot();
    expect(mixins.hideVisually()).toMatchSnapshot();
    expect(mixins.resetButton()).toMatchSnapshot();
    expect(mixins.resetInput()).toMatchSnapshot();
    expect(mixins.resetList()).toMatchSnapshot();
    expect(mixins.resetMedia()).toMatchSnapshot();
    expect(mixins.resetTypography()).toMatchSnapshot();
    expect(mixins.root(darkTheme.var, darkTheme.toTokens().breakpoint)).toMatchSnapshot();
    expect(mixins.textBreak()).toMatchSnapshot();
    expect(mixins.textTruncate()).toMatchSnapshot();
    expect(mixins.textWrap()).toMatchSnapshot();
  });

  it('shadow', () => {
    expect(mixins.shadow(darkTheme.var, 'xs')).toMatchSnapshot();
    expect(mixins.shadow(darkTheme.var, 'sm')).toMatchSnapshot();
    expect(mixins.shadow(darkTheme.var, 'md')).toMatchSnapshot();
    expect(mixins.shadow(darkTheme.var, 'lg')).toMatchSnapshot();
    expect(mixins.shadow(darkTheme.var, 'xl')).toMatchSnapshot();
  });

  it('text', () => {
    expect(mixins.text(darkTheme.var, 'sm')).toMatchSnapshot();
    expect(mixins.text(darkTheme.var, 'df')).toMatchSnapshot();
    expect(mixins.text(darkTheme.var, 'lg')).toMatchSnapshot();
  });
});
