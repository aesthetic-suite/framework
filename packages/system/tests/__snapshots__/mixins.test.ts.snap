// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mixins background errors for invalid \`palette\` 1`] = `"Invalid \\"palette\\" value. Must be one of brand, primary, secondary, tertiary, neutral, muted, danger, warning, failure, success, info"`;

exports[`Mixins background renders background 1`] = `
Object {
  "backgroundColor": "var(--palette-neutral-bg-base)",
}
`;

exports[`Mixins background renders background 2`] = `
Object {
  "backgroundColor": "var(--palette-brand-bg-base)",
}
`;

exports[`Mixins background renders background 3`] = `
Object {
  "backgroundColor": "var(--palette-danger-bg-base)",
}
`;

exports[`Mixins border errors for invalid \`palette\` 1`] = `"Invalid \\"palette\\" value. Must be one of brand, primary, secondary, tertiary, neutral, muted, danger, warning, failure, success, info"`;

exports[`Mixins border errors for invalid \`shade\` 1`] = `"Invalid \\"shade\\" value. Must be one of 00, 10, 20, 30, 40, 50, 60, 70, 80, 90"`;

exports[`Mixins border errors for invalid \`size\` 1`] = `"Invalid \\"size\\" value. Must be one of sm, df, lg"`;

exports[`Mixins border renders border 1`] = `
Object {
  "borderColor": "var(--palette-neutral-color-40)",
  "borderRadius": "var(--border-df-radius)",
  "borderStyle": "solid",
  "borderWidth": "var(--border-df-width)",
}
`;

exports[`Mixins border renders border 2`] = `
Object {
  "borderColor": "var(--palette-info-color-40)",
  "borderRadius": "var(--border-sm-radius)",
  "borderStyle": "solid",
  "borderWidth": "var(--border-sm-width)",
}
`;

exports[`Mixins border renders border 3`] = `
Object {
  "borderColor": "var(--palette-neutral-color-50)",
  "borderStyle": "solid",
  "borderWidth": "var(--border-lg-width)",
}
`;

exports[`Mixins display renders hide completely 1`] = `
Object {
  "display": "none",
}
`;

exports[`Mixins display renders hide offscreen 1`] = `
Object {
  "clipPath": "rect(1px, 1px, 1px, 1px)",
  "height": 1,
  "left": "-5vw",
  "overflow": "hidden",
  "position": "fixed",
  "width": 1,
}
`;

exports[`Mixins display renders hide visually 1`] = `
Object {
  ":not(:focus):not(:active)": Object {
    "border": 0,
    "clip": "rect(0, 0, 0, 0)",
    "clipPath": "inset(50%)",
    "height": 1,
    "margin": 0,
    "overflow": "hidden",
    "padding": 0,
    "position": "absolute",
    "whiteSpace": "nowrap",
    "width": 1,
  },
}
`;

exports[`Mixins foreground errors for invalid \`palette\` 1`] = `"Invalid \\"palette\\" value. Must be one of brand, primary, secondary, tertiary, neutral, muted, danger, warning, failure, success, info"`;

exports[`Mixins foreground renders foreground 1`] = `
Object {
  "color": "var(--palette-neutral-fg-base)",
}
`;

exports[`Mixins foreground renders foreground 2`] = `
Object {
  "color": "var(--palette-success-fg-base)",
}
`;

exports[`Mixins foreground renders foreground 3`] = `
Object {
  "color": "var(--palette-warning-fg-base)",
}
`;

exports[`Mixins heading errors for invalid \`level\` 1`] = `"Invalid \\"level\\" value. Must be one of 1, 2, 3, 4, 5, 6"`;

exports[`Mixins heading renders heading 1`] = `
Object {
  "color": "var(--palette-neutral-fg-base)",
  "fontFamily": "var(--typography-font-heading)",
  "fontSize": "var(--heading-l1-size)",
  "fontWeight": "normal",
  "letterSpacing": "var(--heading-l1-letter-spacing)",
  "lineHeight": "var(--heading-l1-line-height)",
  "wordWrap": "break-word",
}
`;

exports[`Mixins heading renders heading 2`] = `
Object {
  "color": "var(--palette-neutral-fg-base)",
  "fontFamily": "var(--typography-font-heading)",
  "fontSize": "var(--heading-l3-size)",
  "fontWeight": "normal",
  "letterSpacing": "var(--heading-l3-letter-spacing)",
  "lineHeight": "var(--heading-l3-line-height)",
  "wordWrap": "break-word",
}
`;

exports[`Mixins heading renders heading 3`] = `
Object {
  "color": "var(--palette-neutral-fg-base)",
  "fontFamily": "var(--typography-font-heading)",
  "fontSize": "var(--heading-l6-size)",
  "fontWeight": "normal",
  "letterSpacing": "var(--heading-l6-letter-spacing)",
  "lineHeight": "var(--heading-l6-line-height)",
  "wordWrap": "break-word",
}
`;

exports[`Mixins renders interactive 1`] = `
Object {
  ":active": Object {
    "backgroundColor": "var(--palette-neutral-color-50)",
    "borderColor": "var(--palette-neutral-color-60)",
  },
  ":disabled": Object {
    "backgroundColor": "var(--palette-neutral-color-20)",
    "borderColor": "var(--palette-neutral-color-30)",
    "cursor": "default",
  },
  ":focus": Object {
    "borderColor": "var(--palette-neutral-color-60)",
  },
  ":hover": Object {
    "backgroundColor": "var(--palette-neutral-color-60)",
    "borderColor": "var(--palette-neutral-color-70)",
  },
  "backgroundColor": "var(--palette-neutral-color-40)",
  "borderColor": "var(--palette-neutral-color-50)",
  "color": "var(--palette-neutral-color-00)",
  "cursor": "pointer",
}
`;

exports[`Mixins renders interactive 2`] = `
Object {
  ":active": Object {
    "backgroundColor": "var(--palette-primary-color-50)",
    "borderColor": "var(--palette-primary-color-60)",
  },
  ":disabled": Object {
    "backgroundColor": "var(--palette-primary-color-20)",
    "borderColor": "var(--palette-primary-color-30)",
    "cursor": "default",
  },
  ":focus": Object {
    "borderColor": "var(--palette-primary-color-60)",
  },
  ":hover": Object {
    "backgroundColor": "var(--palette-primary-color-60)",
    "borderColor": "var(--palette-primary-color-70)",
  },
  "backgroundColor": "var(--palette-primary-color-40)",
  "borderColor": "var(--palette-primary-color-50)",
  "color": "var(--palette-primary-color-00)",
  "cursor": "pointer",
}
`;

exports[`Mixins reset renders reset button 1`] = `
Object {
  "appearance": "none",
  "backgroundColor": "transparent",
  "border": 0,
  "cursor": "pointer",
  "display": "inline-block",
  "fontSize": "inherit",
  "margin": 0,
  "padding": 0,
  "textDecoration": "none",
  "userSelect": "auto",
  "verticalAlign": "middle",
}
`;

exports[`Mixins reset renders reset button 2`] = `
Object {
  "appearance": "none",
  "backgroundColor": "transparent",
  "border": 0,
  "cursor": "pointer",
  "display": "inline-flex",
  "fontSize": "inherit",
  "margin": 0,
  "padding": 0,
  "textDecoration": "none",
  "userSelect": "auto",
  "verticalAlign": "middle",
}
`;

exports[`Mixins reset renders reset input 1`] = `
Object {
  "::-moz-focus-outer": Object {
    "border": 0,
  },
  "appearance": "none",
  "backgroundColor": "transparent",
  "margin": 0,
  "padding": 0,
  "width": "100%",
}
`;

exports[`Mixins reset renders reset list 1`] = `
Object {
  "listStyle": "none",
  "margin": 0,
  "padding": 0,
}
`;

exports[`Mixins reset renders reset media 1`] = `
Object {
  "display": "block",
  "verticalAlign": "middle",
}
`;

exports[`Mixins reset renders reset typography 1`] = `
Object {
  "fontFamily": "inherit",
  "fontSize": "inherit",
  "fontWeight": "normal",
  "wordWrap": "break-word",
}
`;

exports[`Mixins root renders root 1`] = `
Object {
  "-moz-osx-font-smoothing": "grayscale",
  "-webkit-font-smoothing": "antialiased",
  "@media (min-width: 114.29em) and (max-width: 137.07em)": Object {
    "fontSize": "var(--breakpoint-lg-root-text-size)",
    "lineHeight": "var(--breakpoint-lg-root-line-height)",
  },
  "@media (min-width: 137.14em)": Object {
    "fontSize": "var(--breakpoint-xl-root-text-size)",
    "lineHeight": "var(--breakpoint-xl-root-line-height)",
  },
  "@media (min-width: 45.71em) and (max-width: 68.50em)": Object {
    "fontSize": "var(--breakpoint-xs-root-text-size)",
    "lineHeight": "var(--breakpoint-xs-root-line-height)",
  },
  "@media (min-width: 68.57em) and (max-width: 91.36em)": Object {
    "fontSize": "var(--breakpoint-sm-root-text-size)",
    "lineHeight": "var(--breakpoint-sm-root-line-height)",
  },
  "@media (min-width: 91.43em) and (max-width: 114.21em)": Object {
    "fontSize": "var(--breakpoint-md-root-text-size)",
    "lineHeight": "var(--breakpoint-md-root-line-height)",
  },
  "backgroundColor": "var(--palette-neutral-color-00)",
  "color": "var(--palette-neutral-fg-base)",
  "fontFamily": "var(--typography-font-text)",
  "fontSize": "var(--typography-root-text-size)",
  "lineHeight": "var(--typography-root-line-height)",
  "margin": 0,
  "padding": 0,
  "textRendering": "optimizeLegibility",
  "textSizeAdjust": "100%",
}
`;

exports[`Mixins shadow errors for invalid \`palette\` 1`] = `"Invalid \\"palette\\" value. Must be one of brand, primary, secondary, tertiary, neutral, muted, danger, warning, failure, success, info"`;

exports[`Mixins shadow errors for invalid \`shade\` 1`] = `"Invalid \\"shade\\" value. Must be one of 00, 10, 20, 30, 40, 50, 60, 70, 80, 90"`;

exports[`Mixins shadow errors for invalid \`size\` 1`] = `"Invalid \\"size\\" value. Must be one of xs, sm, md, lg, xl"`;

exports[`Mixins shadow renders shadow 1`] = `
Object {
  "boxShadow": "var(--shadow-md-x) var(--shadow-md-y) var(--shadow-md-blur) var(--shadow-md-spread) var(--palette-neutral-color-90)",
}
`;

exports[`Mixins shadow renders shadow 2`] = `
Object {
  "boxShadow": "var(--shadow-sm-x) var(--shadow-sm-y) var(--shadow-sm-blur) var(--shadow-sm-spread) var(--palette-info-color-90)",
}
`;

exports[`Mixins shadow renders shadow 3`] = `
Object {
  "boxShadow": "var(--shadow-lg-x) var(--shadow-lg-y) var(--shadow-lg-blur) var(--shadow-lg-spread) var(--palette-neutral-color-50)",
}
`;

exports[`Mixins text errors for invalid \`size\` 1`] = `"Invalid \\"size\\" value. Must be one of sm, df, lg"`;

exports[`Mixins text renders text 1`] = `
Object {
  "color": "var(--palette-neutral-fg-base)",
  "fontFamily": "var(--typography-font-text)",
  "fontSize": "var(--text-df-size)",
  "fontWeight": "normal",
  "lineHeight": "var(--text-df-line-height)",
  "wordWrap": "break-word",
}
`;

exports[`Mixins text renders text 2`] = `
Object {
  "color": "var(--palette-neutral-fg-base)",
  "fontFamily": "var(--typography-font-text)",
  "fontSize": "var(--text-lg-size)",
  "fontWeight": "normal",
  "lineHeight": "var(--text-lg-line-height)",
  "wordWrap": "break-word",
}
`;

exports[`Mixins text renders text break 1`] = `
Object {
  "overflowWrap": "break-word",
  "wordBreak": "break-word",
  "wordWrap": "break-word",
}
`;

exports[`Mixins text renders text truncate 1`] = `
Object {
  "overflow": "hidden",
  "textOverflow": "ellipsis",
  "whiteSpace": "nowrap",
}
`;

exports[`Mixins text renders text wrap 1`] = `
Object {
  "overflowWrap": "normal",
  "wordBreak": "normal",
  "wordWrap": "normal",
}
`;

exports[`Mixins ui box renders box 1`] = `
Object {
  "backgroundColor": "var(--palette-neutral-bg-base)",
  "borderColor": "var(--palette-neutral-color-40)",
  "borderRadius": "var(--border-df-radius)",
  "borderStyle": "solid",
  "borderWidth": "var(--border-df-width)",
  "boxShadow": "var(--shadow-md-x) var(--shadow-md-y) var(--shadow-md-blur) var(--shadow-md-spread) var(--palette-neutral-color-90)",
}
`;

exports[`Mixins ui box renders box 2`] = `
Object {
  "backgroundColor": "var(--palette-brand-bg-base)",
  "borderColor": "var(--palette-brand-color-40)",
  "borderRadius": "var(--border-df-radius)",
  "borderStyle": "solid",
  "borderWidth": "var(--border-df-width)",
  "boxShadow": "var(--shadow-md-x) var(--shadow-md-y) var(--shadow-md-blur) var(--shadow-md-spread) var(--palette-neutral-color-90)",
}
`;

exports[`Mixins ui box renders box with custom border options 1`] = `
Object {
  "backgroundColor": "var(--palette-danger-bg-base)",
  "borderColor": "var(--palette-success-color-40)",
  "borderRadius": "var(--border-lg-radius)",
  "borderStyle": "solid",
  "borderWidth": "var(--border-lg-width)",
  "boxShadow": "var(--shadow-md-x) var(--shadow-md-y) var(--shadow-md-blur) var(--shadow-md-spread) var(--palette-neutral-color-90)",
}
`;

exports[`Mixins ui box renders box with custom shadow options 1`] = `
Object {
  "backgroundColor": "var(--palette-brand-bg-base)",
  "borderColor": "var(--palette-brand-color-40)",
  "borderRadius": "var(--border-df-radius)",
  "borderStyle": "solid",
  "borderWidth": "var(--border-df-width)",
  "boxShadow": "var(--shadow-md-x) var(--shadow-md-y) var(--shadow-md-blur) var(--shadow-md-spread) var(--palette-info-color-00)",
}
`;

exports[`Mixins ui box renders box without border 1`] = `
Object {
  "backgroundColor": "var(--palette-neutral-bg-base)",
  "boxShadow": "var(--shadow-md-x) var(--shadow-md-y) var(--shadow-md-blur) var(--shadow-md-spread) var(--palette-neutral-color-90)",
}
`;

exports[`Mixins ui box renders box without shadow 1`] = `
Object {
  "backgroundColor": "var(--palette-brand-bg-base)",
  "borderColor": "var(--palette-brand-color-40)",
  "borderRadius": "var(--border-df-radius)",
  "borderStyle": "solid",
  "borderWidth": "var(--border-df-width)",
}
`;
